packages/php/email-editor/src/Container.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\EmailEditor;
+
+class Container {
+  protected array $services = [];
+  protected array $instances = [];
+
+  public function set(string $name, callable $callable): void {
+    $this->services[$name] = $callable;
+  }
+
+  public function get(string $name) {
```

Comment:
In the free plugin's container we have this annotation for `get`:
```
/**
   * @template T
   * @param class-string<T> $id
   * @return T
   */
```
We use classnames as id and this helps PHPStan and also IDE to figure out type of service when we call `$service = $container->get(Service::class);`
If that works without this annotation please ignore this comment.

Reply:
That's a good point. PHPStan will be added to [the ticket focusing on WP Coding Standard](https://mailpoet.atlassian.net/browse/MAILPOET-6240) so I can add it to the next PR.

Reply:
It's addressed in one of the new commits.